datalad/metadata/metadata.py

```diff
@@ -303,7 +303,7 @@ def _query_aggregated_metadata_singlepath(
     for fpath in [f for f in contentmeta.keys()
                   if rparentpath == curdir or
                   f == rparentpath or
-                  f.startswith(_with_sep(rparentpath))]:
+                  path_startswith(f, rparentpath)]:
```

Comment:
Correct, since `f == rparentpath` was checked in addition. Therefore it can be removed.